src/agent/plugins/build/prepare.ts

```diff
@@ -116,6 +118,46 @@ export function beforeJob(executionContext: cm.IExecutionContext, callback) {
         shell.mkdir('-p', bd);
         shell.cd(bd);
         
+        //
+        // Fix filePath variables to physical paths
+        //
+        trace.write('resolving filePath variables');
+
+        //
+        // Resolve paths for filePath inputs
+        //
+        var job: agentifm.JobRequestMessage = executionContext.jobInfo.jobMessage;
+        job.tasks.forEach((task: agentifm.TaskInstance) => {
+            trace.write('processing task: ' + task.name + ' ( ' + task.id + ')');
+            var taskDef = executionContext.taskDefinitions[task.id];
+            if (!taskDef) {
+                throw new Error('Task definition for ' + task.id + ' not found.');
+            }
+            
+            // find the filePath inputs
+            var filePathInputs: { [key: string]: boolean } = {};
+            taskDef.inputs.forEach((input: agentifm.TaskInputDefinition) => {
+                if (input.type === 'filePath') {
+                    filePathInputs[input.name] = true;
+                    trace.write('filePath input: ' + input.name);                       
+                }
+            });
+            
+            // scan dictionary of input/val for pathInputs
+            for (var key in task.inputs) {
+                if (filePathInputs.hasOwnProperty(key)) {
+                    // defer to the source provider to resolve to the physical path
+                    // default is to append (relative to repo root but tfsvc mapped)
+                    var resolvedPath = scmProvider.resolveInputPath(task.inputs[key] || '');
```

Comment:
This might fail if the input has a variable in it. Lets say, the input = $(build.sourcesdirectory)/sample.sh
It will resolve to " this.targetPath/$(build.sourcesdirectory)/sample.sh" 
path.resolve will treat the input value as a relative path.
May be we should resolve the inputs before doing the expansion?

Below is the error that i get in my repro: 
"Not found SourceFolder: /home/ankit/myagent/vsoxplat/_work/2/s//home/ankit/myagent/vsoxplat/_work/2/s/shell1.sh"